server/src/test/java/org/elasticsearch/gateway/DanglingIndicesStateTests.java

```diff
@@ -181,7 +191,66 @@ public void testDanglingIndicesStripAliases() throws Exception {
         }
     }
 
+    public void testDanglingIndicesAreNotAllocatedWhenDisabled() throws Exception {
+        try (NodeEnvironment env = newNodeEnvironment()) {
+            MetaStateService metaStateService = new MetaStateService(env, xContentRegistry());
+            LocalAllocateDangledIndices localAllocateDangledIndices = mock(LocalAllocateDangledIndices.class);
+
+            final ClusterService mockClusterService = buildMockClusterService(false);
```

Comment:
I would find the test easier to understand if we were to explicitly pass the setting `Settings.builder().put(AUTO_IMPORT_DANGLING_INDICES_SETTING.getKey(), false).build()` here instead of the magic boolean.